extensions/blocks/story/edit.native.js

```diff
@@ -27,123 +26,77 @@ import { icon } from '.';
 import styles from './editor.scss';
 import StoryEditingButton from './story-editing-button';
 
-class StoryEdit extends React.Component {
-	constructor( props ) {
-		super( props );
+const StoryEdit = ( { attributes, isSelected, clientId, setAttributes, onFocus } ) => {
+	const { mediaFiles } = attributes;
+	const hasContent = !! mediaFiles.length;
 
-		this.onEditButtonTapped = this.onEditButtonTapped.bind( this );
+	// setup state vars
+	const [ isUploadInProgress, setUploadInProgress ] = useState( false );
 
-		this.mediaUploadStateReset = this.mediaUploadStateReset.bind( this );
-		this.finishMediaUploadWithSuccess = this.finishMediaUploadWithSuccess.bind( this );
-		this.finishMediaUploadWithFailure = this.finishMediaUploadWithFailure.bind( this );
-		this.updateMediaUploadProgress = this.updateMediaUploadProgress.bind( this );
+	const [ isSaveInProgress, setSaveInProgress ] = useState( false );
 
-		// save progress bindings
-		this.mediaSaveStateReset = this.mediaSaveStateReset.bind( this );
-		this.finishMediaSaveWithSuccess = this.finishMediaSaveWithSuccess.bind( this );
-		this.finishMediaSaveWithFailure = this.finishMediaSaveWithFailure.bind( this );
-		this.updateMediaSaveProgress = this.updateMediaSaveProgress.bind( this );
+	const [ didUploadFail, setUploadFail ] = useState( false );
 
-		this.onStorySaveResult = this.onStorySaveResult.bind( this );
+	const [ didSaveFail, setSaveFail ] = useState( false );
 
-		this.onMediaIdChanged = this.onMediaIdChanged.bind( this );
+	// sync with local media store
+	useEffect( mediaUploadSync, [] );
 
-		this.onStoryPressed = this.onStoryPressed.bind( this );
+	useEffect( mediaSaveSync, [] );
 
-		this.state = {
-			isUploadInProgress: false,
-			isSaveInProgress: false,
-			didUploadFail: false,
-			didSaveFail: false,
-		};
-	}
-
-	isUrlRemote( protocolForUrl ) {
+	function isUrlRemote( protocolForUrl ) {
 		return protocolForUrl === 'http:' || protocolForUrl === 'https:';
 	}
```

Comment:
There is a `prependHTTP` function in `@wordpress/url` that might be used here instead.
Is `isUrlRemote` used at all? 🤔

Reply:
oops you're right! Removed it in d440e25fb4 👍